flocker/dockerplugin/_api.py

```diff
@@ -162,23 +162,18 @@ def got_configured(configured):
         creating.addCallback(lambda _: {u"Err": None})
         return creating
 
-    @app.route("/VolumeDriver.Mount", methods=["POST"])
-    @_endpoint(u"Mount")
-    def volumedriver_mount(self, Name):
+    def _path_response(self, name):
         """
-        Move a volume with the given name to the current node and mount it.
+        Wait until a dataset has been mounted and then return its path.
 
-        Since we need to return the filesystem path we wait until the
-        dataset is mounted locally.
-
-        :param unicode Name: The name of the volume.
+        :param unicode name: The name of the volume.
 
-        :return: Result indicating success.
+        :return: Result that includes the mountpoint.
         """
-        dataset_id = UUID(dataset_id_from_name(Name))
-        d = self._flocker_client.move_dataset(self._node_id, dataset_id)
-
+        # If we ever get rid of dataset_id hashing hack we'll need to
+        # lookup the dataset by its metadata, not its id.
         def get_state():
```

Comment:
I'm not sure how to do this more neatly, given `get_state` is currently the callable in the `deferLater` after this.